### ISSUE TITLE
Issue with Rope Embedding in non Flux mode

### ISSUE THREAD

**Issue by gsp-27** [Kai-46/minFM#5]
Hi,

Wanted to confirm if my understanding is correct before posting a fix. In non Flux mode with lay out being `cos_re1, cos_im1, cos_re2, cos_im2, cos_re3, cos_im3` and similar for sin

When we applying this on q without any `rearrange` we would be pairing wrong in `sin` when `qa, qb = q.chunk(2, dim=-1)` and then `rope_sin * torch.cat([-qb, qa], dim=-1)`

`cos` suggest pairing (x1, x2), (x3, x4) and (x5, x6) but in `sin` pairing suggests (x1, x4), (x2, x5) and (x3, x6)

Easy fix would be not applying `if` clause here and always returning `(cos_re1, cos_re2, cos_re3, cos_im1, cos_im2, cos_im3)` and same for `sin`.

Then in flux mode we are pairing (x1, x2), (x3, x4), (x5, x6) and in non flux mode we are pairing (x1, x4), (x2, x5) and (x3, x6)
